test-data/unit/check-typevar-tuple.test

```diff
@@ -94,3 +94,71 @@ reveal_type(g(args))  # N: Revealed type is "Tuple[builtins.str, builtins.str, b
 reveal_type(h(args))  # N: Revealed type is "Tuple[builtins.str, builtins.str, builtins.int]"
 [builtins fixtures/tuple.pyi]
 
+[case testTypeVarTupleGenericClassDefn]
+from typing import Generic, TypeVar, Tuple
+from typing_extensions import TypeVarTuple
+
+T = TypeVar("T")
+Ts = TypeVarTuple("Ts")
+
+class Variadic(Generic[Ts]):
+    pass
+
+class Mixed1(Generic[T, Ts]):
+    pass
+
+class Mixed2(Generic[Ts, T]):
+    pass
+
+variadic: Variadic[int, str]
+reveal_type(variadic)  # N: Revealed type is "__main__.Variadic[builtins.int, builtins.str]"
+
+variadic_single: Variadic[int]
+reveal_type(variadic_single)  # N: Revealed type is "__main__.Variadic[builtins.int]"
+
+empty: Variadic[()]
+# TODO: fix pretty printer to be better.
+reveal_type(empty)  # N: Revealed type is "__main__.Variadic"
+
+m1: Mixed1[int, str, bool]
+reveal_type(m1)  # N: Revealed type is "__main__.Mixed1[builtins.int, builtins.str, builtins.bool]"
+
+[builtins fixtures/tuple.pyi]
+
+[case testTypeVarTupleGenericClassWithFunctions]
+from typing import Generic, Tuple, TypeVar
+from typing_extensions import TypeVarTuple, Unpack
+
+Ts = TypeVarTuple("Ts")
+T = TypeVar("T")
+
+class Variadic(Generic[T, Ts]):
```

Comment:
This test fails if I add a suffix type variable, with a weird error `Argument 1 to "foo" has incompatible type "Variadic[int, str, bool, float]"; expected "Variadic[int, str, bool, float]" (diff)`

Reply:
I left a comment below about what seems to be going on.